test-data/unit/check-class-namedtuple.test

```diff
@@ -407,7 +407,7 @@ class HasNone(NamedTuple):
     x: int
     y: Optional[int] = None
 
-reveal_type(HasNone(1))  # E: Revealed type is 'Tuple[builtins.int, builtins.int, fallback=__main__.HasNone]'
+reveal_type(HasNone(1))  # E: Revealed type is 'Tuple[builtins.int, Union[builtins.int, builtins.None], fallback=__main__.HasNone]'
```

Comment:
It is in most cases, actually.  Mypy has 2 different ways of rendering types: an internal representation and an external representation.  The external representation is what's used for all errors, and renders `Union[X, None]` as `Optional[X]`.  The internal representation is used by `reveal_type` for some reason that I no longer recall, and does not do this translation.  I've been wanting to clean this up since forever (and make them all have a bit more PEP 484 style), but haven't ever had the time.